daemon/k8s_watcher.go

```diff
@@ -1421,3 +1457,49 @@ func (d *Daemon) updateCiliumNetworkPolicyV2(ciliumV2Store cache.Store,
 	d.deleteCiliumNetworkPolicyV2(oldRuleCpy)
 	d.addCiliumNetworkPolicyV2(ciliumV2Store, newRuleCpy)
 }
+
+func (d *Daemon) updateK8sPodV1(oldK8sPod, newK8sPod *v1.Pod) {
+	if oldK8sPod == nil || newK8sPod == nil {
+		return
+	}
+
+	// We only care about label updates
+	oldPodLabels := oldK8sPod.GetLabels()
+	newPodLabels := newK8sPod.GetLabels()
+	if reflect.DeepEqual(oldPodLabels, newPodLabels) {
+		return
+	}
+	eps := endpointmanager.GetEndpoints()
+
+	podNSName := k8sUtils.GetObjNamespaceName(&newK8sPod.ObjectMeta)
+	var podEP *endpoint.Endpoint
+	for _, ep := range eps {
```

Comment:
Add a new function `endpointmanager.LookupPodName()` that uses `endpointmanager.lookupPodNameLocked` as we already maintain a hash table to access endpoint via `namespace:podname`.